client/src/clock-register/clock-register.directive.spec.js

```diff
@@ -44,20 +46,43 @@ describe('dojo clock register directive', function () {
         expect(list[0]).toBeDefined();
       });
 
-      it('after clicking the button, should show the time in the list', function () {
-        var firstItem;
+      describe('after clicking the button', function () {
+        it('in case of success, should show the time in the list', function () {
+          var firstItem;
 
-        $httpBackend.expect('POST', '/api/v1/clock/hit').respond(200, {time: '16:20:00'});
-        hitThePointBtn.click();
-        $httpBackend.flush();
+          $httpBackend.expect('POST', '/api/v1/clock/hit').respond(200, {time: '16:20:00'});
+          hitThePointBtn.click();
+          $httpBackend.flush();
 
-        firstItem = element.find('#daily-list :eq(0)');
+          firstItem = element.find('#daily-list :eq(0)');
 
-        expect(firstItem.text()).toBe('16:20:00');
-      });
+          expect(firstItem.text()).toBe('16:20:00');
+        });
+
+        it('if session is lost, should warn the user with message', function () {
+          var feedbackMessage;
+          $httpBackend.expect('POST', '/api/v1/clock/hit').respond(401, 'Authentication Failed');
```

Comment:
Yes! I think we should stick to objects or only direct body strings. Not mixing both!